cpp/include/cudf/io/text/data_chunk_source_factories.hpp

```diff
@@ -190,18 +234,18 @@ class file_data_chunk_source : public data_chunk_source {
 };
 
 /**
- * @brief a host string data source which creates an istream_data_chunk_reader
+ * @brief a host string data source which creates an host_span_data_chunk_reader
  */
-class string_data_chunk_source : public data_chunk_source {
+class host_span_data_chunk_source : public data_chunk_source {
  public:
-  string_data_chunk_source(std::string const& data) : _data(data) {}
+  host_span_data_chunk_source(host_span<const char> data) : _data(data) {}
   [[nodiscard]] std::unique_ptr<data_chunk_reader> create_reader() const override
   {
-    return std::make_unique<istream_data_chunk_reader>(std::make_unique<std::istringstream>(_data));
+    return std::make_unique<host_span_data_chunk_reader>(_data);
```

Comment:
Good catch here. I didn't realize this was performing a copy, but it makes sense in retrospect. How else would `istringstream` be assured the string is kept alive? :man_shrugging: